vertx-web-openapi/src/main/java/io/vertx/ext/web/openapi/AuthenticationHandlerProvider.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2021 Red Hat, Inc.
+ *
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *  The Eclipse Public License is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  The Apache License v2.0 is available at
+ *  http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.ext.web.openapi;
+
+import io.vertx.codegen.annotations.Fluent;
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.handler.AuthenticationHandler;
+import io.vertx.ext.web.openapi.impl.AuthenticationHandlerProviderImpl;
+
+import java.util.function.Function;
+
+/**
+ * An authentication handler provider. This class will hold factories for creating {@link AuthenticationHandler}
+ * objects.
+ *
+ * @author <a href="mailto:plopes@redhat.com">Paulo Lopes</a>
+ */
+@VertxGen
+public interface AuthenticationHandlerProvider {
```

Comment:
Would it be better to have this in vertx-web?

Reply:
No, this was a helper. After your feedback and reworking the API I've a less convoluted API.